rholang/src/main/scala/coop/rchain/rholang/interpreter/PrettyPrinter.scala

```diff
@@ -174,16 +180,29 @@ case class PrettyPrinter(freeShift: Int, boundShift: Int) {
   private def buildPattern(patterns: Seq[Channel]): (Int, String) =
     ((0, "") /: patterns.zipWithIndex) {
       case ((patternsFree, string), (pattern, i)) =>
-        (patternsFree + freeCount(pattern), string + buildString(pattern) + {
-          if (i != patterns.length - 1) ", "
-          else ""
-        })
+        (patternsFree + freeCount(pattern),
+          string +
+            this.copy(currentId = nextId).buildString(pattern) + {
```

Comment:
that's not right.
When you build a string for a pattern:
PrettyPrinter(freeshift=boundShift, boundShift=0, freePrefix=boundPrefix, boundPrefix = boundPrefix + 1.)

I can't make heads or tails of nextId vs currentId, please rename them.